dashboard/app/models/script.rb

```diff
@@ -1055,6 +1055,8 @@ def summarize(include_stages = true, user = nil)
 
     has_older_course_progress = course.try(:has_older_version_progress?, user)
     has_older_script_progress = has_older_version_progress?(user)
+    user_script = user && UserScript.find_by(user: user, script: self)
```

Comment:
We should use the existing [user_scripts association](https://github.com/code-dot-org/code-dot-org/blob/7b2080acd2c123e1f6fbb2df6388d555bb1327b6/dashboard/app/models/script.rb#L41):

```ruby
user_script = user_scripts.find_by(user: user)
```

Note since [UserScript.user](https://github.com/code-dot-org/code-dot-org/blob/da1ec09d868056f163078506884aaaf818c9bd5a/dashboard/app/models/user_script.rb#L6) is required, not null, the above will return `nil` whenever `user` is `nil` (since there are no matches), so we don't need the pre-check.